astropy/table/info.py

```diff
@@ -0,0 +1,85 @@
+"""
+Table method and functions related to providing information about
+columns and tables.
+"""
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import sys
+import os
+
+import numpy as np
+
+__all__ = ['info']
+
+def info(self, option='attributes', out=''):
+    """
+    Write summary information about column to the ``out`` filehandle.
+    By default this prints to standard output via sys.stdout.
+
+    The ``option` argument specifies what type of information
+    to include.  This can be a string, a function, or a list of
+    strings or functions.  Built-in options are:
+
+    - ``meta``: basic column meta data like ``dtype`` or ``format``
+    - ``stats``: basic statistics: minimum, mean, and maximum
```

Comment:
Here's an example:
```
t = Table.read('http://fermi.gsfc.nasa.gov/ssc/data/access/lat/4yr_catalog/gll_psc_v16.fit')
t.info(option='stats')
/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/numpy/core/_methods.py:102: RuntimeWarning: invalid value encountered in subtract
  x = asanyarray(arr - arrmean)
<Table length=3034>
        name             min          mean          std          max    
-------------------- ----------- ------------- ------------- -----------
         Source_Name          --            --            --          --
             RAJ2000      0.0377       185.638       100.585     359.881
             DEJ2000    -87.6185      -2.23514       41.5371     88.7739
                GLON    0.126161       186.273       107.572     359.969
                GLAT    -87.9516       1.22728       36.4998     86.3679
   Conf_68_SemiMajor         nan           nan           nan         nan
   Conf_68_SemiMinor         nan           nan           nan         nan
      Conf_68_PosAng         nan           nan           nan         nan
   Conf_95_SemiMajor         nan           nan           nan         nan
   Conf_95_SemiMinor         nan           nan           nan         nan
      Conf_95_PosAng         nan           nan           nan         nan
             ROI_num           1 424.777191826 238.846231996         840
          Signif_Avg        -inf          -inf           nan     1048.96
        Pivot_Energy     100.798       2104.46        2578.2     37202.0
        Flux_Density 4.21515e-16   1.29389e-11   2.17231e-10 1.14476e-08
    Unc_Flux_Density        -inf          -inf           nan 6.86476e-10
```
Stilts gives this:
```
$ stilts tpipe fermi.fits omode=stats
Total Rows: 3034
+----------------------+----------------+---------------+---------------+---------------+------+
| column               | mean           | stdDev        | min           | max           | good |
+----------------------+----------------+---------------+---------------+---------------+------+
| Source_Name          |                |               |               |               | 3034 |
| RAJ2000              | 185.63786      | 100.58523     | 0.0377        | 359.8812      | 3034 |
| DEJ2000              | -2.235143      | 41.53712      | -87.6185      | 88.7739       | 3034 |
| GLON                 | 186.27281      | 107.57225     | 0.12616104    | 359.9692      | 3034 |
| GLAT                 | 1.2272775      | 36.499836     | -87.95155     | 86.36791      | 3034 |
| Conf_68_SemiMajor    | -Infinity      |               | -Infinity     | 0.6587264     | 3031 |
| Conf_68_SemiMinor    | -Infinity      |               | -Infinity     | 0.32094488    | 3031 |
| Conf_68_PosAng       | -0.21967068    | 51.926346     | -89.94        | 89.92         | 3006 |
| Conf_95_SemiMajor    | -Infinity      |               | -Infinity     | 1.0681        | 3031 |
| Conf_95_SemiMinor    | -Infinity      |               | -Infinity     | 0.5204        | 3031 |
| Conf_95_PosAng       | -0.21967068    | 51.926346     | -89.94        | 89.92         | 3006 |
| ROI_num              | 424.7772       | 238.84624     | 1             | 840           | 3034 |
| Signif_Avg           | -Infinity      |               | -Infinity     | 1048.9587     | 3034 |
| Pivot_Energy         | 2104.4626      | 2578.198      | 100.79754     | 37201.97      | 3034 |
| Flux_Density         | 1.29388835E-11 | 2.1723061E-10 | 4.2151523E-16 | 1.1447582E-8  | 3034 |
| Unc_Flux_Density     | -Infinity      |               | -Infinity     | 6.864756E-10  | 3034 |
```
I think putting min / max next to each other is nicer.

Handing nan / inf in the stats computations is tricky , but generally I think having a column like "good" that prints what `np.isfinite` gives would be very helpful.